creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -27,11 +31,64 @@ const SyncType = Object.freeze({
 
 // Phases in `issueUpdateReplicaSetOp`. Used for debugging if method fails
 const issueUpdateReplicaSetOpPhases = Object.freeze({
-  GET_HEALTHY_CONTENT_NODES: 'GET_HEALTHY_CONTENT_NODES',
+  DETERMINE_NEW_REPLICA_SET: 'DETERMINE_NEW_REPLICA_SET',
   ENQUEUE_SYNCS: 'ENQUEUE_SYNCS',
   UPDATE_URSM_REPLICA_SET: 'UPDATE_URSM_REPLICA_SET'
 })
 
+// Modes used in issuing a reconfig. Each successive mode is a superset of the mode prior.
+// The `key` of the reconfig states is used to identify the current reconfig mode.
+// The `value` of the reconfig states is used in the superset logic of determining which type of
+// reconfig is enabled.
+const RECONFIG_MODE_KEYS = [
+  // Reconfiguration is entirely disabled
+  'RECONFIG_DISABLED',
+  // Reconfiguration is enabled only if one secondary is unhealthy
+  'ONE_SECONDARY',
+  // Reconfiguration is enabled for one secondary and multiple secondaries (currently two secondaries)
+  'MULTIPLE_SECONDARIES',
+  // Reconfiguration is enabled for one secondary, multiple secondaries, a primary, and a primary and one secondary
+  'PRIMARY_AND_OR_SECONDARIES',
+  // Reconfiguration is enabled for one secondary, multiple secondaries, a primary, and a primary and one secondary,
+  // and entire replica set
+  // Note: this mode will probably be disabled.
+  'ENTIRE_REPLICA_SET'
+]
+
+/*
+  `RECONFIG_MODES` will look like:
+  {
+    RECONFIG_DISABLED: {
+      key: 'RECONFIG_DISABLED',
+      value: 0
+    },
+    ONE_SECONDARY: {
+      key: 'ONE_SECONDARY',
+      value: 1
+    },
+    MULTIPLE_SECONDARIES: {
+      key: 'MULTIPLE_SECONDARIES',
+      value: 2
+    },
+    PRIMARY_AND_OR_SECONDARIES: {
+      key: 'PRIMARY_AND_OR_SECONDARIES',
+      value: 3
+    },
+    ENTIRE_REPLICA_SET: {
+      key: 'ENTIRE_REPLICA_SET',
+      value: 4
+    }
+  }
+*/
+let RECONFIG_MODES = {}
+RECONFIG_MODE_KEYS.forEach((mode, i) => {
+  RECONFIG_MODES[mode] = {
+    key: mode,
+    value: i
+  }
+})
+RECONFIG_MODES = Object.freeze(RECONFIG_MODES)
```

Comment:
eh my take was if we were to ever add modes, we can  add `RECONFIG_MODES` and then dynamically create it with the appropriate level. I can keep it hard coded since we want to be careful with the modes anyway. refactoring